src/main/_docs/integration-guide/integration-version-control.md

```diff
@@ -25,6 +25,53 @@ For other git-based or SVN-based repository providers, please refer to their doc
 ## GitLab oAuth
 Currently it's not possible for Codenvy to use oAuth integration with GitLab. Although GitLab supports oAuth for clone operations, pushes are not supported. You can track [this GitLab issue](https://gitlab.com/gitlab-org/gitlab-ce/issues/18106) in their issue management system.
 
+## BitBucket Server oAuth
+### Setup oAuth for BitBucket Server
+#### 1. Generate Keys
+SSH to your BitBucket Server instance and generate RSA key-pairs
+```shell
+# generate the key pairs
+$ openssl genrsa -out private.pem 1024
+
+# copy the key and pem to a location where it can be accessed
+$ openssl rsa -in private.pem -pubout > public.pub`
+$ openssl pkcs8 -topk8 -inform pem -outform pem -nocrypt -in private.pem -out privatepkcs8.pem 
+```
+#### 2. Setup Codenvy in BitBucket Server
+In the Bitbucket Server as an Admin:
+
+1. Go to Administration -> Application Links
+1. Enter your Codenvy URL in the 'application url' field and press the 'Create new link' button.
+1. When  the 'Configure Application URL' window appears press the 'Continue' button.
+1. In the ‘Link applications’ window fill in the following fields and press the 'Continue' button:
+  - Application Name : Codenvy
+  - Service Provider Name : Codenvy
+  - Consumer key : `generated by BBS` (Save this string, you will need it later)
+  - Shared secret : `generated by BBS`
+  - Request Token URL : {your Bitbucket Server URL}/plugins/servlet/oauth/request-token
+  - Access token URL : {your Bitbucket Server URL}/plugins/servlet/oauth/access-token
+  - Authorize URL : {your Bitbucket Server URL}/plugins/servlet/oauth/authorize
+
+#### 3. Add Codenvy oAuth Info
+In the following screen press the pencil icon to edit and select the 'Incoming Authentication' menu. Fill in the following fields in the ‘OAuth’ tab:
+- Consumer Key : {the consumer key from the previous step}
+- Consumer Name : Codenvy
+- Public Key : {the key from public.pub file} (The key must not contain ‘-----BEGIN PUBLIC KEY-----’ and ‘-----END PUBLIC KEY-----’ rows)
+- Consumer Callback : {your Codenvy URL>}/api/oauth/1.0/callback
+
+#### 4. Configure Codenvy Properties
+In the `codenvy.pp` file add (or change if they already exist):
+- bitbucket_consumer_key : the consumer key you have entered before
+- bitbucket_private_key : the key from privatepkcs8.pem file 
+- (The key must not contain ‘----BEGIN PRIVATE KEY-----’ and ‘-----END PRIVATE KEY-----’ rows) (The key must be specified as a single raw)
+- bitbucket_endpoint : replace the default Bitbucket url by your Bitbucket Server url
+
+#### 5. Restart Codenvy
+Restart your Codenvy server to enable the integration.
+
+### Using BitBucket Server oAuth
+With oAuth setup you can perform all git actions with the repo, including commits and pushes. You can also use the Codenvy pull request panel (found on the far right of pull-request enabled workspaces).
```

Comment:
This is not true, OAuth1 allows only to acces the Bitbucket Server API. Remote Git operations wil work only by ssh.